internal/acceptance/device_helpers.go

```diff
@@ -0,0 +1,119 @@
+package acceptance
```

Comment:
Not requesting any changes, just commenting:

These helpers may turn out to all be "metal" related. In the future we may want to consider moving such service-shared testing / utility code to a service specific home.

internal/services/metal/{resourcename}
internal/services/metal/testing

More likely, and for more gain, I think we would want to move towards a testing pattern like the one in the Linode provider:
https://github.com/linode/terraform-provider-linode/tree/dev/linode/instance/tmpl